book/src/transaction-fees.md

```diff
@@ -0,0 +1,52 @@
+# Deterministic Transaction Fees
+
+Transactions currently include a fee field that indicates the maximum fee field
+a slot leader is permitted to charge to process a transaction. The cluster, on
+the other hand, agrees on a minimum fee. If the network is congested, the slot
+leader may prioritize the transactions offering higher fees. That means the
+client won't know how much was collected until the transaction is confirmed by
+the cluster and the remaining balance is checked. It smells of exactly we
+dislike about Ethereum's "gas", non-determinism.
+
+## Implementation Status
+
+This design is not yet implemented, but is written as though it has been.  Once
+implemented, delete this comment.
+
+### The Fees Program
+
+The Solana cluster has an on-chain program called Fees with just one global
+account called the *fees account*. The account holds a set of fee parameters
+that the cluster will use to calculate transaction fees in the following epoch.
+Any slot leader may submit congestion statistics to the fees account and the
+Fees program uses that data to automatically adjust the fees to maximize
+resource usage.  Each validator checks congestion statistics and rejects blocks
+that include congestion data inconsistent with its own observations.
```

Comment:
If we just base it on the ledger data, we could weed out all potential consistencies.  But I was thinking about the network bandwidth available. If one validator has a 1 mbps connection to the cluster, it'd want to crank up the fees so that traffic is low enough that it can participate.  Likewise, someone trying to centralize the network could get a 100 gbps connection, force fees extremely low, such that traffic jumps, and none of the slower validators can participate.